### PR TITLE
Implement multi-stream source test adapter

### LINKED ISSUE #372: Implement multi-stream source test adapter
The adapter must create multiple video streams and send them with different source ids like `source-%d`. We can specify the parallelism and number of frames. The adapter is used for performance tests. At the end of the operation, it sends a shutdown if configured (#371).

### PR DESCRIPTION
**Pull request by tomskikh**

### CODE REVIEW

File: adapters/gst/gst_plugins/python/savant_rs_serializer.py [hunk 1]
```diff
@@ -127,6 +129,37 @@ class SavantRsSerializer(LoggerMixin, GstBase.BaseTransform):
...
+            'Enable multistream',
+            'Enable multistream',
+            False,
+            GObject.ParamFlags.READWRITE | Gst.PARAM_MUTABLE_READY,
+        ),
+        'source-id-pattern': (
+            str,
+            'Pattern for source ID',
+            'Pattern for source ID when multistream is enabled. E.g. "source-%d".',
+            None,
```
Comment (denisvmedyantsev) [inline adapters/gst/gst_plugins/python/savant_rs_serializer.py]: Default should be `source-%d`

File: adapters/gst/gst_plugins/python/savant_rs_serializer.py [hunk 2]
```diff
@@ -237,11 +283,24 @@ def do_set_property(self, prop: GObject.GParamSpec, value: Any):
...
+        elif prop.name == 'number-of-streams':
+            self.number_of_streams = value
+        elif prop.name == 'shutdown-auth':
+            self.shutdown_auth = value
         else:
             raise AttributeError(f'Unknown property {prop.name}.')
 
     def do_start(self):
-        assert self.source_id, 'Source ID is required.'
+        if not self.enable_multistream:
```
Comment (denisvmedyantsev) [inline adapters/gst/gst_plugins/python/savant_rs_serializer.py]: The pattern is required for multistream, it is better to check.

File: adapters/gst/gst_plugins/python/savant_rs_serializer.py [hunk 3]
```diff
@@ -391,6 +466,23 @@ def build_video_frame(
...
+                self.source_id_pattern % i for i in range(self.number_of_streams)
+            ]
+            if len(source_ids) != len(set(source_ids)):
+                raise ValueError(
+                    'Duplicate source IDs. Check source-id-pattern property.'
+                )
+        elif self.source_id is not None:
+            source_ids = [self.source_id]
+        else:
+            source_ids = []
```
Comment (denisvmedyantsev) [inline adapters/gst/gst_plugins/python/savant_rs_serializer.py]: `source_ids = []` - Is it a valid case?

**Review by denisvmedyantsev — APPROVED** [review]
Go through my comments. No need for a second review.

File: adapters/gst/gst_plugins/python/savant_rs_serializer.py (hunk 1 above)
Comment (tomskikh) [reply adapters/gst/gst_plugins/python/savant_rs_serializer.py]: Set a default value to property description.

File: adapters/gst/gst_plugins/python/savant_rs_serializer.py (hunk 2 above)
Comment (tomskikh) [reply adapters/gst/gst_plugins/python/savant_rs_serializer.py]: Added a check.

File: adapters/gst/gst_plugins/python/savant_rs_serializer.py (hunk 3 above)
Comment (tomskikh) [reply adapters/gst/gst_plugins/python/savant_rs_serializer.py]: Moved content of `_set_source_ids_and_zmq_sockets` to `do_start` and removed this case.

**Pull request merged**